src/strands/multiagent/a2a/server.py

```diff
@@ -88,18 +91,33 @@ def public_agent_card(self) -> AgentCard:
             capabilities=self.capabilities,
         )
 
-    @property
-    def agent_skills(self) -> list[AgentSkill]:
-        """Get the list of skills this agent provides.
+    def _get_skills_from_tools(self) -> list[AgentSkill]:
+        """Get the list of skills from Strands agent tools.
 
         Skills represent specific capabilities that the agent can perform.
         Strands agent tools are adapted to A2A skills.
 
         Returns:
             list[AgentSkill]: A list of skills this agent provides.
         """
-        # TODO: translate Strands tools (native & MCP) to skills
-        return []
+        return [
+            AgentSkill(name=config["name"], id=config["name"], description=config["description"], tags=[])
+            for config in self.strands_agent.tool_registry.get_all_tools_config().values()
+        ]
+
+    @property
+    def agent_skills(self) -> list[AgentSkill]:
+        """Get the list of skills this agent provides."""
+        return self._agent_skills
```

Comment:
I think to keep this up to date, this needs to call the getter on-demand:

```suggestion
        return self._agent_skills or self._get_skills_from_tools()
```

And then in the constructor do not call `self._get_skills_from_tools` - just store the value if we have one.

That way the behavior is:
  - If you set skills, then use those
  - else get the skills from tools

Reply:
Is there a use case where you'd want an empty set of tags (`[]`)? in which case this should check explicitly for `None` instead of falsey - since I believe an empty python array is falsey